### PR TITLE
bugfix: create etcd object in `xpcall`, this step may fail

### PR DESCRIPTION
**Pull request by membphis**
fix issue: #2310

1. The old process, if creating etcd fails, etcd data will no longer be synchronized.  We need to create the etcd object in `xpcall`. 
2. the return value should be `res, err` of `etcd.new`.

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

**Comment by membphis** [comment]
@Yiyiyimu please take a look at this issue, it is a bugfix

**Comment by Yiyiyimu** [comment]
@membphis Sure that's my mistake. Thank you for the fix!!

**Comment by nic-chen** [comment]
I checked the old version. we created a etcd cli only once here: 
https://github.com/apache/apisix/blob/7f64ba810ef4ea9a105f7c3e8a83ac54e47ad58e/apisix/core/config_etcd.lua#L434-L446
and then we could reuse it.

I think we could use that mode.

do you remember that change it? @Yiyiyimu

**Comment by membphis** [comment]
> do you remember that change it? @Yiyiyimu

I know the reason. 

When we create a new etcd v3 object, it depends on `cosocket` feature. so we have to move this code block into `timer` land.

**Comment by membphis** [comment]
I just updated the initialization process and now it only initializes once. https://github.com/apache/apisix/pull/2312/commits/bd2d95038d29593c1b205fc4516768ea94bd4976

**Comment by nic-chen** [comment]
> > do you remember that change it? @Yiyiyimu
> 
> 
> 
> I know the reason. 
> 
> 
> 
> When we create a new etcd v3 object, it depends on `cosocket` feature. so we have to move this code block into `timer` land. it is not supports `cosocket` feature in `init` and `init_worker` phase.

OK

**Review by nic-chen — APPROVED** [review]

**Comment by moonming** [comment]
@membphis please choose a better title of this PR

**Comment by membphis** [comment]
> @membphis please choose a better title of this PR

updated

File: apisix/core/config_etcd.lua [hunk 1]
```diff
@@ -416,16 +416,20 @@ local function _automatic_fetch(premature, self)
...
         local ok, err = xpcall(function()
+            if not self.etcd_cli then
+                local etcd_cli, err = etcd.new(self.etcd_conf)
+                if not etcd_cli then
+                    error("failed to create etcd instance for key ["
+                          .. self.key .. "]: " .. (err or "unknown"))
+                end
+                self.etcd_cli = etcd_cli
+            end
+
```
Comment (liuhengloveyou) [inline apisix/core/config_etcd.lua:432]: Why has the number of returned values changed?

Comment (membphis) [reply apisix/core/config_etcd.lua:432]: the old code is wrong. the return value is `res, err` for latest version of `resty-etcd`

**Review by Yiyiyimu — APPROVED** [review]

**Pull request merged**